src/L2GovernorMetadata.sol

```diff
@@ -28,7 +38,21 @@ abstract contract L2GovernorMetadata {
     virtual
   {
     _proposals[proposalId] = Proposal(voteStart, voteEnd, isCanceled);
-    emit ProposalAdded(proposalId, voteStart, voteEnd, isCanceled);
+    if (isCanceled) {
+      emit ProposalCanceled(proposalId);
```

Comment:
Agree with this if/else approach. Does what we want even though it feels a bit odd 👍